Public/Invoke-Locksmith.ps1

```diff
@@ -162,6 +186,23 @@
         $CAHosts | ForEach-Object { $SafeUsers += '|' + $_.Name }
     }
 
+    if ( $Scans ) {
+        $Results = Invoke-Scans -Scans $Scans
+            # Re-hydrate the findings arrays from the Results hash table
+            $AllIssues      = $Results['AllIssues']
+            $AuditingIssues = $Results['AuditingIssues']
+            $ESC1           = $Results['ESC1']
+            $ESC2           = $Results['ESC2']
+            $ESC3           = $Results['ESC2']
+            $ESC4           = $Results['ESC4']
+            $ESC5           = $Results['ESC5']
+            $ESC6           = $Results['ESC6']
+            $ESC8           = $Results['ESC8']
+    }
+    else {
+        # Possibly pull the checks below into this block so the scans don't all run a second time.
```

Comment:
You are correct, the checks in lines 206-229 should be removed. If it's not, everything runs a second time! Eek!